File: app/src/main/java/com/example/lanhuajian/blues/module_kaiyan/detail/OEDetailRelatedViewHolder.java
package com.example.lanhuajian.blues.module_kaiyan.detail;

import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.lanhuajian.blues.R;
import com.example.lanhuajian.blues.framework.utils.TimeFormatUtil;
import com.example.lanhuajian.blues.module_kaiyan.OpenEyeEntity;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;

/**
 * User : Blues
 * Date : 2019/3/6
 */

public class OEDetailRelatedViewHolder extends BaseViewHolder<OEDetailRelatedEntity.ItemListBean> {

    private LinearLayout itemText, itemVideo;
    private TextView tvText, tvTitle;
    private ImageView ivCover;

    public OEDetailRelatedViewHolder(ViewGroup parent) {
        super(parent, R.layout.item_oe_detail_related);

        itemText = $(R.id.ll_item_text);
        itemVideo = $(R.id.ll_item_video);

        tvText = $(R.id.tv_text);
        tvTitle = $(R.id.tv_title);

        ivCover = $(R.id.iv_cover);

        itemText.setVisibility(View.GONE);
        itemVideo.setVisibility(View.GONE);
    }

    @Override
    public void setData(OEDetailRelatedEntity.ItemListBean data) {
        super.setData(data);

        if (TextUtils.equals(data.getType(), "textCard")) {
            itemText.setVisibility(View.VISIBLE);
            itemVideo.setVisibility(View.GONE);
            tvText.setText(data.getData().getText());

        } else if (TextUtils.equals(data.getType(), "videoSmallCard")) {
            itemText.setVisibility(View.GONE);
            itemVideo.setVisibility(View.VISIBLE);
            Glide.with(getContext()).load(data.getData().getCover().getFeed()).into(ivCover);
            tvTitle.setText(data.getData().getTitle());
        }
    }
}

File: app/src/main/java/com/blues/kaiyan/detail/view/OEDetailRelatedViewHolder.kt
package com.blues.kaiyan.detail.view

import android.view.ViewGroup
import com.blues.kaiyan.detail.model.OEDetailRelatedEntity
import com.blues.R
import android.widget.LinearLayout
import android.widget.TextView
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.jude.easyrecyclerview.adapter.BaseViewHolder

/**
 * User : Blues
 * Date : 2019/3/6
 */
class OEDetailRelatedViewHolder(parent: ViewGroup?) :
    BaseViewHolder<OEDetailRelatedEntity.ItemListBean>(parent, R.layout.item_oe_detail_related) {

    private val itemText = itemView.findViewById<LinearLayout>(R.id.ll_item_text).apply {
        visibility = View.GONE
    }
    private val itemVideo = itemView.findViewById<LinearLayout>(R.id.ll_item_video).apply {
        visibility = View.GONE
    }
    private val tvText: TextView = itemView.findViewById(R.id.tv_text)
    private val tvTitle: TextView = itemView.findViewById(R.id.tv_title)
    private val ivCover: ImageView = itemView.findViewById(R.id.iv_cover)

    override fun setData(data: OEDetailRelatedEntity.ItemListBean) {
        super.setData(data)

        if (TextUtils.equals(data.type, "textCard")) {
            itemText.visibility = View.VISIBLE
            itemVideo.visibility = View.GONE
            tvText.text = data.data.text
        } else if (TextUtils.equals(data.type, "videoSmallCard")) {
            itemText.visibility = View.GONE
            itemVideo.visibility = View.VISIBLE
            Glide.with(context).load(data.data.cover.feed).into(ivCover)
            tvTitle.text = data.data.title
        }
    }
}